src/sphinxfilter.cpp

```diff
@@ -385,16 +385,31 @@ class FilterString_c : public IFilter_Str
 	bool Eval ( const CSphMatch & tMatch ) const override
```

Comment:
I'd make a new `class FilterStringCpm_c : public FilterString_c ` and left the code in the base class as is for the `EStrCmpDir::EQ` . That also eliminates the switch on the most use path of the string equality comparison. I'd override `Eval` function in the derived class for all other comparison functions.

or you could make template from `FilterString_c` and specialize its comparison function based on the `eStrCmpDir`

Reply:
Is there a concern with efficiency, or do you think that's a better abstraction than current? Or maybe this is to avoid introducing unintended behavior?

Reply:
it will:
- keep current code as is
- not affect speed of `=` string comparison

Reply:
I see your point with trying to avoid modifying existing lines to avoid unintended behavior. However, I don't think speed should be a concern. We cannot avoid introducing a branch anyway - this refactor just moves the branch into `CreateFilter`:

```
...
else if ( tSettings.m_eStrCmpDir==EStrCmpDir::EQ )
			return std::make_unique<FilterString_c> ( eCollation, tSettings.m_bHasEqualMin || tSettings.m_bHasEqualMax );
		else
			return std::make_unique<FilterStringCmp_c> ( eCollation, tSettings.m_eStrCmpDir );
```

Reply:
I've modified this PR so as to avoid modifying lines - let me know if this looks good.

Reply:
I suggest this change [filter_string.patch](https://github.com/manticoresoftware/manticoresearch/files/14474606/filter_string.patch) for `FilterString_c` inheritance or something similar - do not copy all code but derive from the `FilterString_c` and override only function that you really need to change or as I said you could make template from FilterString_c and specialize its `Eval` method based on the eStrCmpDir

Reply:
Done